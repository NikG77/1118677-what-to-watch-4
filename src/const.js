export const RatingLevel = {
  BAD_MIN: 0,
  NORMAL_MIN: 3,
  GOOD_MIN: 5,
  VERY_GOOD_MIN: 8,
  AWESOME: 10,
};

export const TIME_DELAY = 1000;

export const TabName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};


export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
  ERROR: `ERROR`,
  BAD_REQUEST: `BAD_REQUEST`,
};


export const Error = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400
};

