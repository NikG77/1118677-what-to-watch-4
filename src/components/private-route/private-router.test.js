import React from "react";
import renderer from "react-test-renderer";
import PrivateRouter from "./private-route.jsx";
import history from "../../history.js";
import {Router} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {AuthorizationStatus} from "../../const.js";

const mockStore = configureStore([]);


it(`PrivateRouter component render correctly`, () => {
  const store = mockStore({
    [NameSpace.WATCH]: {
      genre: `All genres`,
      movieCount: 4,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      isAuthorizationLoading: false,
    },
    [NameSpace.DATA]: {
      allMovies: [],
    },
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRouter
              render={() => {}}
              path={``}
              exact={true}
              isAuthorizationStatus={false}
              isAuthorizationLoading={false}
            />
          </Router>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
