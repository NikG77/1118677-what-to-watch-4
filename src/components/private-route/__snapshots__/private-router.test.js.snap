// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouter component render correctly 1`] = `null`;
